src/server/Join.js

```diff
@@ -66,10 +66,20 @@ export default class Join {
       (now - this._getLastPublishedTime() >= this.interval);
   }
 
-  removeContext({ _subscriptionId }) {
-    this.contexts = this.contexts.filter(
-      context => context._subscriptionId !== _subscriptionId,
-    );
+  _findContextIndex({ _subscriptionId }) {
+    return this.contexts.findIndex(c => c._subscriptionId === _subscriptionId);
```

Comment:
_Just a comment, updating not required_
It would be safer if you check for `c` not nullable first. My old code also does not check for that, but overtime I have found that it is generally safer to have a check, because we never what would happen when concurrency involves.

Reply:
If we get into real concurrent code (not only callbacks and all the new stuff based on it like promises or async-await), I guess we would have to rethink a lot of code we wrote.

Reply:
Yeah, I think more checks for `nullable` values and be careful with the `worker` should be enough